test/rules/assertions/validTypes.js

```diff
@@ -329,5 +410,52 @@ export default {
           }
       `,
     },
+    {
+      code: `
+          /**
+           * @constant {string}
+           */
+           const FOO = 'foo';
+      `,
+    },
+    {
+      code: `
+          /**
+           * @constant {string} FOO
+           */
+           const FOO = 'foo';
+      `,
+    },
+    {
+      code: `
+          /**
+           * @extends Foo
+           */
+           class Bar {};
+      `,
+    },
+    {
+      code: `
+          /**
+           * @extends {Foo<String>}
+           */
+           class Bar {};
+      `,
+    },
```

Comment:
I.e., for this test, this should only pass for `settings.jsdoc.mode` set to "gcc" or "typescript".